src/Compilers/CSharp/Portable/Symbols/Source/FieldSymbolWithAttributesAndModifiers.cs

```diff
@@ -382,6 +382,11 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
         {
             base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
 
+            if (this.RefKind == RefKind.RefReadOnly)
```

Comment:
The call to `CSharpCompilation.EnsureIsReadOnlyAttributeExists()` from `SourceFieldSymbol` during binding is necessary to ensure that the symbol for the well-known `IsReadOnlyAttribute` is created if missing and marked as embedded in the assembly. Without this, the call to `PEModuleBuilder.SynthesizeIsReadOnlyAttribute()` will fail during emit since the attribute declaration is missing.